File: app/src/main/java/tr/com/cherrysunshinysky/todonotesapp/adapter/NotesAdapter.java
package tr.com.cherrysunshinysky.todonotesapp.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import tr.com.cherrysunshinysky.todonotesapp.R;
import tr.com.cherrysunshinysky.todonotesapp.clicklistener.ItemClickListener;
import tr.com.cherrysunshinysky.todonotesapp.model.Notes;

/**
 * Created by Emir U. Özen on 2/27/2021.
 * emir.ozen@outlook.com
 */
public class NotesAdapter extends RecyclerView.Adapter<NotesAdapter.ViewHolder> {

    List<Notes> listNotes;
    ItemClickListener itemClickListener;

    public NotesAdapter(List<Notes> list, ItemClickListener itemClickListener) {
        this.listNotes = list;
        this.itemClickListener = itemClickListener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.notes_adapter_layout, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull NotesAdapter.ViewHolder holder, int position) {
        Notes notes = listNotes.get(position);
        String description = notes.getDescription();
        String title = notes.getTitle();

        holder.textViewTitle.setText(title);
        holder.textViewDescription.setText(description);
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                itemClickListener.onClick();
            }
        });
    }

    @Override
    public int getItemCount() {
        return listNotes.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        TextView textViewTitle, textViewDescription;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            textViewTitle = itemView.findViewById(R.id.tv_title);
            textViewDescription = itemView.findViewById(R.id.tv_description);
        }
    }
}

File: app/src/main/java/tr/com/cherrysunshinysky/todonotesapp/adapter/NotesAdapter.kt
package tr.com.cherrysunshinysky.todonotesapp.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import tr.com.cherrysunshinysky.todonotesapp.R
import tr.com.cherrysunshinysky.todonotesapp.clicklistener.ItemClickListener
import tr.com.cherrysunshinysky.todonotesapp.model.Notes

/**
 * Created by Emir U. Özen on 3/13/2021
 * emir.ozen@outlook.com
 */
class NotesAdapter(private var listNotes: List<Notes>,private var itemClickListener: ItemClickListener) : RecyclerView.Adapter<NotesAdapter.ViewHolder>() {

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var textViewTitle: TextView = view.findViewById(R.id.tv_title)
        var textViewDescription: TextView = view.findViewById(R.id.tv_description)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.notes_adapter_layout, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val notes = listNotes[position]
        val title = notes.title
        val description = notes.description
        holder.textViewTitle.text = title
        holder.textViewDescription.text = description
        holder.itemView.setOnClickListener {
            itemClickListener.onClick(notes)
        }
    }

    override fun getItemCount() = listNotes.size
}